src/hotspot/share/opto/macro.cpp

```diff
@@ -3232,11 +3217,9 @@ bool PhaseMacroExpand::expand_macro_nodes() {
     case Node::Class_CallStaticJava:
       expand_mh_intrinsic_return(n->as_CallStaticJava());
       C->remove_macro_node(n);
-      assert(C->macro_count() == (old_macro_count - 1), "expansion must have deleted one node from macro list");
       break;
     case Node::Class_FlatArrayCheck:
       expand_flatarraycheck_node(n->as_FlatArrayCheck());
-      assert(C->macro_count() == (old_macro_count - 1), "expansion must have deleted one node from macro list");
```

Comment:
Why did you delete these asserts?

Reply:
It seems the checking happens nevertheless right after switch:

https://github.com/openjdk/valhalla/blob/5e88e8a08c4532840c66177e0032953ba6e2956c/src/hotspot/share/opto/macro.cpp#L3220-L3227

Reply:
Right, good catch!